test/IonTimestamp.ts

```diff
@@ -242,4 +242,11 @@ describe("Timestamp", () => {
         assert.deepEqual(ts.getSecondsDecimal(), ion.Decimal.parse('45'));
         assert.deepEqual(ts._getFractionalSeconds(), ion.Decimal.ZERO);
     });
+    it('constructor with Date', () => {
+        const date = new Date(2000, 0, 1, 12, 30, 45);
+        let timestamp1 = new ion.Timestamp(-7 * 60, 2000, 1, 1, 12, 30, 45);
+        let timestamp2 = ion.Timestamp.fromDate(date);
+        assert.equal(timestamp1!.equals(timestamp2!), timestamp1 === timestamp2);  // data model equivalence
```

Comment:
This assertion is a bit off, which is causing this test to pass despite the offset discrepancy described above.

If `timestamp1` and `timestamp2` are not equal to each other, then both
```typescript
timestamp1!.equals(timestamp2!)
```
and
```typescript
timestamp1 === timestamp2!
```
will be `false`. That means that `assert.equal` will be satisfied and the test will pass, which I don't believe is the intent.

I think this can be simplified to:
```typescript
assert.isTrue(timestamp1!.equals(timestamp2!));
```